toolchain/lex/lex.cpp

```diff
@@ -683,10 +687,12 @@ static auto DispatchNext(Lexer& lexer, llvm::StringRef source_text,
         source_text[position])](lexer, source_text, position);
   }
 
-  // Incomplete ranges will use the next token for their end; we want that to be
-  // `FileEnd` in this case, so check before adding `FileEnd`. The argument is
-  // just the final character for diagnostic locations.
-  lexer.EndDumpSemIRRangeIfIncomplete(source_text.end() - 1);
+  if (lexer.has_dump_sem_ir_ranges()) {
+    // Incomplete ranges will use the next token for their end; we want that to
+    // be `FileEnd` in this case, so check before adding `FileEnd`. The argument
+    // is just the final character for diagnostic locations.
+    lexer.EndDumpSemIRRangeIfIncomplete(source_text.end() - 1);
```

Comment:
Can we use `source_text.end()` here instead? I think that's the location we normally use for the end of the file, rather than using the final character in the file.

Reply:
Done, I'd incorrectly assumed this would be a bounds issue.

Reply:
Nevermind, it *is* a bounds issue. I think we typically don't point diagnostics at the end of the file, because autoupdate can't handle this.

Reply:
(added a TODO, but I've got too many things on my plate right now)